gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -264,4 +280,29 @@ protected String getTokenDisplayText(final String token) {
     return String.format(Locale.ROOT, "%s...%s", token.substring(0, 10), token.substring(token.length() - 3));
   }
 
+  /**
+   * Method that deletes expired tokens based on the token timestamp.
+   */
+  protected void evictExpiredTokens() {
+    try {
+      for (final String token : getTokens()) {
+        if (isExpired(token)) {
```

Comment:
I've been wondering if we should allow some "grace period" beyond the expiration if renewal is still possible for a given token before evicting it.

Reply:
Hmm, wouldn't that make the token expiration time a bit ambiguous? are you suggesting that renew token should be possible until (expiry + x) seconds?

Reply:
Yes, as we've discussed, while renewers likely will request renewals prior to expiration, they are not currently required to do so. Delaying the eviction avoids potential "unknown token" issues for renewers, and does not affect the expiration time.

Reply:
Thanks for explaining Phil, fixed it in new PR.